### PR TITLE
Revisions

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+| **Purge on delete** | Configure whether to delete the reivions upon document delete, or create a delete marker instead. |
+| **Limit # of revisions** | Configure how much revisions to keep. Default: unlimited. |
+| **Limit # of revisions by age** | Configure a minimum retention time before the revisions can be expired. Default: None. |
+| **Disable** | If true, disalbe the reivions feature for this configuration (default or specific collection). Defalut: false. |
+
+You can also configure the revisions feature using the client:
+
+{CODE configuration@Server\Revisions.cs /}
+
+It is possible to have a default configuration telling the revisions feature to revision all documents (`Disabled=false` on the defulat configuration) and only keep up to 5 revisions, purging older ones (`MinimumRevisionsToKeep=5`).
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: "`Disabled=false` on the defulat configuration" - I don't know if that's disabled or not by default. We need to be more clear here

Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: I don't think you need to put related C# properties (`MinimumRevisionsToKeep=5` and `Disabled=false`). Simple, clear text will work here

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Than we override the behavior of the reivions feature by specifing a configuration specifically to a collection. For example, let's say that we don't want to revision users.
+
+Conversely, we can disable the default configuration (`Disalbed = true`) but configure a collection to enable reivions.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: but configure a collection to enable reivions. -> but enable revisions for specific collections ?

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+As you can see, we have full audit record of all the changes that were made to the document.
+
+You can access the revisions of a specific document by the document's ID ("users/1").
+Or you can access a specific revision by its change vector.
+
+{CODE get_revisions@Server\Revisions.cs /}
+
+Now, let's delete the document. 
+The document would be removed the document but a revision will be created, so you aren't going to lose the audit trail if the document is deleted.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: The document would be removed the document but a revision will be created -> The document would be removed but a revision will be created ?

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+As you can see, we have full audit record of all the changes that were made to the document.
+
+You can access the revisions of a specific document by the document's ID ("users/1").
+Or you can access a specific revision by its change vector.
+
+{CODE get_revisions@Server\Revisions.cs /}
+
+Now, let's delete the document. 
+The document would be removed the document but a revision will be created, so you aren't going to lose the audit trail if the document is deleted. 
+The new revision can be accessed using the studio under the Revisions Bin:
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: I understand what you meant here but "new revision can be accessed using the studio under the Revisions Bin" might be confusing to the users -> new revision can be found in Revisions Bin ?

Please explain what is Revisions Bin

**Comment by arekpalinski** [comment]
BTW there is a lot of typos here, you can add VS spell checker extension to avoid them

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown (hunk 1 above)
Comment (fitzchak) [reply Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: In the configuration options table we list Disabled with the default of false.
I'm made it more clear.
See the previous version of the documentation, we also had those code example, as I say that you can do something but need to show how.

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown (hunk 2 above)
Comment (fitzchak) [reply Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: Changed

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown (hunk 3 above)
Comment (fitzchak) [reply Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: Thanks

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown (hunk 4 above)
Comment (fitzchak) [reply Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: Done

**Comment by fitzchak** [comment]
ReSpeller does not find here typos.

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+The revisions feature will create a revision (snapshot) for every document, upon every update received, or when it is deleted.
+It is useful when you need to track the history of the documents or when you need a full audit trail.
+
+## Configuration
+
+You can configure the revisions feature using the studio:
+
+![Configuring revisions feature on the database](images/configure-revisions.png)
+
+By default, the reivions feature will track history for all documents and never purge old revisions.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: reivions -> revisions

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+## Configuration
+
+You can configure the revisions feature using the studio:
+
+![Configuring revisions feature on the database](images/configure-revisions.png)
+
+By default, the reivions feature will track history for all documents and never purge old revisions. 
+You can configure this for all collections (using the default configuration) and you can have a configuration for a specific collection.
+
+Configuration optoins:
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: optoins -> options

File: Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+| Configuration option | Description |
+| **Purge on delete** | Configure whether to delete the reivions upon document delete, or create a delete marker instead. |
+| **Limit # of revisions** | Configure how much revisions to keep. Default: unlimited. |
+| **Limit # of revisions by age** | Configure a minimum retention time before the revisions can be expired. Default: None. |
+| **Disabled** | If true, disalbe the reivions feature for this configuration (default or specific collection). Defalut: false. |
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/revisions.markdown]: Defalut -> Default

**Comment by arekpalinski** [comment]
ReSpeller checks only .cs files IIRC. I use `Visual Studio Spell Checker (VS2017 and later)`.

I commented 3 typos out of 11

**Pull request merged**